docs/forwarding_logs_to_gateway.md

```diff
@@ -0,0 +1,119 @@
+
+2. Create a `ServiceAccount` to generate the `Secret` which will be used to authorize the forwarder.
+
+```console
+kubectl -n openshift-logging create serviceaccount <SERVICE_ACCOUNT_NAME>
+```
+
+3. Configure the forwarder and deploy it to the `openshift-logging` namespace.
+
+4. Create the following `ClusterRole` and `ClusterRoleBinding` which will allow the cluster to authenticate the user(s) submitting the logs:
+
+```yaml
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: lokistack-dev-tenant-logs-role
+rules:
+- apiGroups:
+  - 'loki.openshift.io'
+  resources:
+  - application
+  - infrastructure
+  - audit
+  resourceNames:
+  - logs
+  verbs:
+  - 'get'
+  - 'create'
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: lokistack-dev-tenant-logs-binding
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: lokistack-dev-tenant-logs-role
+subjects:
+- kind: User
+  name: "kube:admin"
+- kind: ServiceAccount
+  name: "<SERVICE_ACCOUNT_NAME>"
+  namespace: openshift-logging
+```
+
+### Promtail
+
+[Promtail](https://grafana.com/docs/loki/latest/clients/promtail/) is an agent managed by Grafana which forwards logs to a Loki instance. The Grafana documentation can be consulted for [configuring](https://grafana.com/docs/loki/latest/clients/promtail/configuration/#configuration-file-reference) and [deploying](https://grafana.com/docs/loki/latest/clients/promtail/installation/#kubernetes) an instance of Promtail in a Kubernetes cluster.
+
+To configure Promtail to send application, audit, and infrastructure logs, add the following clients to the Promtail configuration
+
+```yaml
+clients:
+  - # ...
+    bearer_token_file: /var/run/secrets/kubernetes.io/serviceaccount/token
+    url: http://lokistack-gateway-http-lokistack-dev.openshift-logging.svc:8080/api/logs/v1/audit/loki/api/v1/push
+  - # ...
+    bearer_token_file: /var/run/secrets/kubernetes.io/serviceaccount/token
+    url: http://lokistack-gateway-http-lokistack-dev.openshift-logging.svc:8080/api/logs/v1/application/loki/api/v1/push
+  - # ...
+    bearer_token_file: /var/run/secrets/kubernetes.io/serviceaccount/token
+    url: http://lokistack-gateway-http-lokistack-dev.openshift-logging.svc:8080/api/logs/v1/infrastructure/loki/api/v1/push
+```
+
+The rest of the configuration can be configured to the developer's desire.
+
+### Fluentd
+
+Loki can receive logs from Fluentd via the [Grafana plugin](https://grafana.com/docs/loki/latest/clients/fluentd/).
+
+The Fluentd configuration can be overrided to target the `application` endpoint to send those log types.
+
+```
+<match **>
+  @type loki
+  # ...
+  bearer_token_file: /var/run/secrets/kubernetes.io/serviceaccount/token
+  url: http://lokistack-gateway-http-lokistack-dev.openshift-logging.svc:8080/api/logs/v1/application
+</match>
+```
+
+## Troubleshooting
+
+### Log Entries Out of Order
```

Comment:
This depends highly on the t-shirt size you pick. IIRC `1x.medium` can handle `IngestionRate: 10` and `IngestionBurstSize: 20`. Right?